rabbitmq-operator/src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/UserReconciler.java

```diff
@@ -44,60 +41,53 @@ public UserReconciler(
     }
 
     public void reconcile(final RabbitMQCluster cluster) {
-        final RabbitMQConnectionInfo connectionInfo = new RabbitMQConnectionInfo(
-                cluster.getName(),
-                cluster.getNamespace(),
-                RabbitMQServices.getDiscoveryServiceName(cluster.getName())
-        );
-
         final Map<String, RabbitMQUser> expectedUsers = cluster.getUsers().stream().collect(Collectors.toMap(RabbitMQUser::getUsername, user -> user));
-        final Map<String, User> existingUsers;
-        final RabbitManagementApi apiClient = managementApiProvider.getApi(connectionInfo);
-        try {
-            existingUsers = RabbitApiResponseConsumer.consumeResponse(apiClient.listUsers().execute())
-                    .stream()
-                    .collect(Collectors.toMap(User::getName, user -> user));
-        } catch (final Exception e) {
-            throw new RuntimeException(String.format("Failed to retrieve existing users from cluster %s in namespace %s, so skipping user reconciliation for this cluster", cluster.getName(), cluster.getNamespace()), e);
-        }
+        final RabbitManagementApiFacade apiClient = managementApiProvider.getApi(cluster);
+
+        final Map<String, User> existingUsers = apiClient.listUsers()
```

Comment:
I feel like there's a better name than 'existingUsers', since it's really 'existing users with write privileges'.

Reply:
hmmm, i think READ_ONLY_USERS is misleading then, since this isn't only existing users with write privileges :)